### ISSUE TITLE
How to set title when playing local HLS playlist file

### ISSUE THREAD

**Issue by tathastu871** [Rendevior/mpv-android-argparse#6]
tried mpv url --title="title"
not working

**Comment by Rendevior** [comment]
> tried mpv url --title="title" not working

In android:
Use `--media-title="{your_title}"`

For some reason title flag doesnt work on android. But that `media-title` both works on different platforms.

**Comment by tathastu871** [comment]
but how to pass title via adb intent
vlc support am -e title option.
how to same in mpv

**Comment by tathastu871** [comment]
can you give example.
not working

> > tried mpv url --title="title" not working
> 
> In android: Use `--media-title="{your_title}"`
> 
> For some reason title flag doesnt work on android. But that `media-title` both works on different platforms.

**Comment by tathastu871** [comment]
how to playlocal m3u8 or media file termux. tried
am -e filepath option but not working

**Comment by fearocanity** [comment]
> how to playlocal m3u8 or media file termux. tried am -e filepath option but not working

You can play local files using `file://`, i will fix this in PR.

> can you give example. not working
> 
> > > tried mpv url --title="title" not working
> > 
> > 
> > In android: Use `--media-title="{your_title}"`
> > For some reason title flag doesnt work on android. But that `media-title` both works on different platforms.

I updated the readme, make sure you inserted the correct ones.
It would look something like this, just like you're in a normal mpv
```
mpv --media-title="Hello World" file://file.mp4
```
Although this only works on files stored in your `phone storage` and also `sdcard`, not on your Home dir and so on.

**Comment by Rendevior** [comment]
> how to playlocal m3u8 or media file termux. tried am -e filepath option but not working

You can play local files using `file://`, i will fix this in PR.

> can you give example. not working
> 
> > > tried mpv url --title="title" not working
> > 
> > 
> > In android: Use `--media-title="{your_title}"`
> > For some reason title flag doesnt work on android. But that `media-title` both works on different platforms.

I updated the readme, make sure you inserted the correct ones.
It would look something like this, just like you're in a normal mpv
```
mpv --media-title="Hello World" file://file.mp4
```
Although this only works on files stored in your `phone storage` and also `sdcard`, not on your Home dir and so on.

**Comment by Rendevior** [comment]
But theres a neat way to do it by using `netcat` to serve your files to be accessible in MPV via http local. I will just merge this into the pr i created to fully fix this filepath problems.

**Comment by tathastu871** [comment]
> > how to playlocal m3u8 or media file termux. tried am -e filepath option but not working
> 
> That extra args is no dice in my side either, I can play local files only using `file://`.
> 
> > can you give example. not working
> > > > tried mpv url --title="title" not working
> > > 
> > > 
> > > In android: Use `--media-title="{your_title}"`
> > > For some reason title flag doesnt work on android. But that `media-title` both works on different platforms.
> 
> I updated the readme, make sure you inserted the correct mpv config setup
> 
> ```
> include="/storage/emulated/0/mpv/mpv.tmp.mp4"
> ```
> 
> It would look something like this, just like you're in a normal mpv
> 
> ```
> mpv -nc --media-title="Hello World" file://file.mp4
> ```
> 
> ~Although this only works on files stored in your `phone storage` and also `sdcard`, not on your Home dir and so on.~ Edit: And in the latest commit, i added netcat to support media files in termux home directory without having root or any sorts.

file://file.mp4 ?? what path it is sdcard or termux

is file.mp4 in your example location is /sdcard/file.mp4 ??

tried file://sdcard/file.webm
but it opens and closes instantly


Also media title not working for urls??.
eg mpv --media-title="title" http://somesite.m3u8

**Comment by Rendevior** [comment]
> > > how to playlocal m3u8 or media file termux. tried am -e filepath option but not working
> > 
> > 
> > That extra args is no dice in my side either, I can play local files only using `file://`.
> > > can you give example. not working
> > > > > tried mpv url --title="title" not working
> > > > 
> > > > 
> > > > In android: Use `--media-title="{your_title}"`
> > > > For some reason title flag doesnt work on android. But that `media-title` both works on different platforms.
> > 
> > 
> > I updated the readme, make sure you inserted the correct mpv config setup
> > ```
> > include="/storage/emulated/0/mpv/mpv.tmp.mp4"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > It would look something like this, just like you're in a normal mpv
> > ```
> > mpv -nc --media-title="Hello World" file://file.mp4
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ~Although this only works on files stored in your `phone storage` and also `sdcard`, not on your Home dir and so on.~ Edit: And in the latest commit, i added netcat to support media files in termux home directory without having root or any sorts.
> 
> file://file.mp4 ?? what path it is sdcard or termux
> 
> is file.mp4 in your example location is /sdcard/file.mp4 ??
> 
> tried file://sdcard/file.webm but it opens and closes instantly
> 
> Also media title not working for urls??. eg mpv --media-title="title" http://somesite.m3u8

No, i'm doing it in the same directory where your target file is..
And also `file://sdcard/file.webm` is wrong... This must be `file:///sdcard/file.webm`.

Because if the `file://` was stripped, it would result that sdcard is where the same directory as you were on not on the `/` directory.

`file://` <- a protocol
`/sdcard/file.webm` <- is where the file at.

![Screenshot_20230209_154335](https://user-images.githubusercontent.com/93582751/217738127-8cbb743c-d064-46af-b19b-2e4832e56ad1.JPG)
![Screenshot_20230209_154308](https://user-images.githubusercontent.com/93582751/217738158-a648124e-4fa7-48a4-940b-7ac8e09bba05.jpg)

**Comment by tathastu871** [comment]
using github latet version of debugbale mpv

tried your netcat commit stopped playing video with error 
mpv  --media-title="hello" file:///sdcard/ty.mp4
mpv: /storage/emulated/0/ty.mp4: No such file or directory

**Comment by Rendevior** [comment]
> media title not working. filename appers as media titlenot custom
> 
>  Screen_Recording_20230209-195220_One.UI.Home.mp4

I see, your version seems unupdated with `fix` branch, That's why its not working.
Try updating it first:
```
curl -sL -o "$PREFIX/bin/mpv" "https://raw.githubusercontent.com/Rendevior/mpv-android-argparse/fix/mpv"
```

**Comment by tathastu871** [comment]
> > media title not working. filename appers as media titlenot custom
> > Screen_Recording_20230209-195220_One.UI.Home.mp4
> 
> I see, your version seems to be outdated with `fix` branch, That's why its not working. Try updating it first:
> 
> ```
> curl -sL -o "$PREFIX/bin/mpv" "https://raw.githubusercontent.com/Rendevior/mpv-android-argparse/fix/mpv"
> ```

tried but title not working

**Comment by Rendevior** [comment]
> > > media title not working. filename appers as media titlenot custom
> > > Screen_Recording_20230209-195220_One.UI.Home.mp4
> > 
> > 
> > I see, your version seems to be outdated with `fix` branch, That's why its not working. Try updating it first:
> > ```
> > curl -sL -o "$PREFIX/bin/mpv" "https://raw.githubusercontent.com/Rendevior/mpv-android-argparse/fix/mpv"
> > ```
> 
> tried but title not working

Have you checked your `mpv.conf`?
Make sure to set it as:
```
include="/storage/emulated/0/mpv/mpv.tmp.mp4"
```

**Comment by tathastu871** [comment]
include="/storage/emulated/0/mpv/mpv.tmp.mp4"
added already

**Comment by tathastu871** [comment]
hey https://github.com/pystardust/ani-cli
this script sets custom title to mpv-android player.
can you read and decipher what technique it was using to set title

**Comment by Rendevior** [comment]
> hey https://github.com/pystardust/ani-cli this script sets custom title to mpv-android player. can you read and decipher what technique it was using to set title

Uhmm, thats odd, on their android player `mpv` doesn't seem to play it with title. Try removing the `mpv` on your `$PATH`, see if still works. Also indicate your logs to point out whats happening.
